os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -206,35 +191,161 @@ binmgr_uinfo_t *binary_manager_get_udata(uint32_t bin_idx)
 }
 
 /****************************************************************************
- * Name: binary_manager_register_ubin
+ * Name: binary_manager_register_upart
  *
  * Description:
- *	 This function registers user binaries.
+ *	 This function registers a partition of user binaries.
  *
  ****************************************************************************/
-int binary_manager_register_ubin(char *name, uint32_t version, uint8_t load_priority)
+void binary_manager_register_upart(char *name, int part_num, int part_size)
 {
-	if (name == NULL || g_bin_count >= USER_BIN_COUNT) {
-		bmdbg("ERROR: Invalid parameter\n");
-		return ERROR;
+	int bin_idx;
+
+	if (part_num < 0 || part_size <= 0 || g_bin_count >= USER_BIN_COUNT) {
+		bmdbg("ERROR: Invalid part info : num %d, size %d\n", part_num, part_size);
```

Comment:
You checked `g_bin_count` also.
How about printing this value also?

Reply:
Good. I added printing `g_bin_count`